blog/2022/05-30-resoto-metrics/index.md

```diff
@@ -0,0 +1,351 @@
+
+```
+avg_over_time(resoto_instances_total{status="running"}[1h])
+```
+
+![Prometheus summed metrics](img/prometheus_avg_over_time.png)
+
+Good, the data points are connected and averaged over time. However the amount of labels is a bit overwhelming. Right now we are seeing one stacked chart per unique label combination. Let's try to reduce the amount of labels by summing them all up.
+
+```
+
+sum(avg_over_time(resoto_instances_total{status="running"}[1h]))
+
+```
+
+![Prometheus summed metrics](img/prometheus_summed_metrics.png)
+
+Nice, now we see how the total number of compute instances has changed over the last two weeks. However we lost absolutely all labels. No more accounts, region and instance type information. To get some information back, let's group the summed up averages by account.
+
+```
+
+sum(avg_over_time(resoto_instances_total{status="running"}[1h])) by (account)
+
+```
+
+![Prometheus summed metrics by account](img/prometheus_summed_metrics_by_account.png)
+
+Neat, we see how the number of compute instances has changed over time for each account.
+
+Want to see how storage has changed over time? Just change `resoto_instances_total` to `resoto_volume_bytes`. Want to see $$$ spent per hour? `resoto_instances_hourly_cost_estimate` is the metric you are looking for.
+
+## How Metrics Are Made
+
+Now, the Prometheus web UI will provide syntax help and auto-complete for available metric names. However, you might be wondering—how are you supposed to know which metrics exist? How do you know what other metrics there are and where something like `resoto_instances_total` is defined?
+
+Metrics are configured in the `resoto.metrics` [configuration](/docs/getting-started/configuring-resoto). Within [Resoto Shell (`resh`)](/docs/concepts/components/shell), simply execute:
+
+```bash
+> config edit resoto.metrics
+```
+
+```yaml
+resotometrics:
+  metrics:
+    instances_total:
+      # Metric help text
+      help: 'Number of Instances'
+      # Aggregation search to run
+      search: 'aggregate(/ancestors.cloud.reported.name as cloud, /ancestors.account.reported.name as account, /ancestors.region.reported.name as region, instance_type as type, instance_status as status: sum(1) as instances_total): is(instance)'
+      # Type of metric (gauge or counter)
+      type: 'gauge'
+...
+```
+
+Here you can add your own metrics. As [previously explained](#aggregating), the `aggregate` expression in the `search` field is what creates the samples of a metric.
+
+This config can be updated at runtime. Next time the `metrics` [workflow](/docs/concepts/automation/workflow) is run Resoto Metrics will generate the new metric and from then on provide it to Prometheus.
+
+```bash
+> workflow run metrics
+```
+
+## Creating the Metrics Dashboard
+
+So now we have learned how to get metrics from Resoto into Prometheus, how to query them and how to create new ones. But what about the dashboard?
+
+Right, fasten your seatbelt. This will go fast.
```

Comment:
```suggestion
Now that we've learned how to get metrics from Resoto into [Prometheus](https://prometheus.io), query metrics, and define new metrics, we can create the dashboard.

Alright, fasten your seatbelts! This will go fast. 🏎️💨
```